dark/filter.py

```diff
@@ -116,6 +116,93 @@ def accept(self, title):
         return self.DEFAULT_ACCEPT
 
 
+class SequenceFilter:
+    """
+    Provide an acceptance test for sequences.
+
+    @param whitelist: If not C{None}, a C{set} of exact sequences that are always
+        acceptable.
+    @param blacklist: If not C{None}, a C{set} of exact sequences that are never
+        acceptable.
+    @param whitelistFile: If not C{None}, a C{str} filename containing lines
+        that give exact sequences that are always acceptable.
+    @param blacklistFile: If not C{None}, a C{str} filename containing lines
+        that give exact sequences that are never acceptable.
+    @param positiveRegex: If not C{None}, a C{str} regex that sequence sequences
+        must match (case is ignored).
+    @param negativeRegex: If not C{None}, a C{str} regex that sequence sequences
+        must not match (case is ignored).
+    """
+
+    REJECT = 0
+    WHITELIST_ACCEPT = 1
+    DEFAULT_ACCEPT = 2
+
+    def __init__(
+        self,
+        whitelist=None,
+        blacklist=None,
+        whitelistFile=None,
+        blacklistFile=None,
+        positiveRegex=None,
+        negativeRegex=None,
+    ):
+        whitelist = whitelist or set()
+        if whitelistFile:
+            with open(whitelistFile) as fp:
+                for line in fp:
+                    whitelist.add(line[:-1])
```

Comment:
Do you need to strip off `\n`s from the ends of line?

Reply:
That's what the `[:-1]` does.  Probably I should have used `strip` but this is a copy of old code. I'll change it.